src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,53 @@
+package duke.task;
+
+/**
+ * Represents a task that can be marked as complete.
+ */
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Marks the {@link Task} as complete.
+     */
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Returns true if the {@link Task} is complete, otherwise returns false.
+     *
+     * @return true if the {@link Task} is complete, otherwise returns false.
+     */
+    public boolean isDone() {
+        return isDone;
+    }
+
+    /**
+     * Returns a tick or X symbol.
+     *
+     * @return a tick or X symbol.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "\u2713" : "\u2718"); // return tick or X symbols
+    }
+
+    @Override
+    public String toString() {
+        return "[" + getStatusIcon() + "] " + description;
+    }
+
+    /**
+     * Returns a stringified version of the {@link Task} that is meant to be used in the {@link duke.storage.Storage}.
+     *
+     * @return a stringified version of the {@link Task} that is meant to be used in the {@link duke.storage.Storage}.
+     */
+    public String stringify() {
```

Comment:
The method name `stringify` is good. However, I feel that in the context where this method is called, `stringify` doesn't seem to convey what the method actually does. 😅

Reply:
Haha i agree, I was trying to think of a better name but resorted to using stringify following how JSON did it https://www.w3schools.com/js/js_json_stringify.asp
After looking through again though, i realise i should make it a static method i.e. Storage.stringify(Task) instead.